docs/CORE_DEV_GUIDE.md

```diff
@@ -0,0 +1,144 @@
+
+### How to Conduct A Good Review
+
+*Always* be kind to contributors. Contributors are often doing
+volunteer work, for which we are tremendously grateful. Provide
+constructive criticism on ideas and implementations, and remind
+yourself of how it felt when your own work was being evaluated as a
+novice.
+
+`napari` strongly values mentorship in code review.  New users
+often need more handholding, having little to no git
+experience. Repeat yourself liberally, and, if you don’t recognize a
+contributor, point them to our development guide, or other GitHub
+workflow tutorials around the web. Do not assume that they know how
+GitHub works (e.g., many don't realize that adding a commit
+automatically updates a pull request). Gentle, polite, kind
+encouragement can make the difference between a new core developer and
+an abandoned pull request.
+
+When reviewing, focus on the following:
+
+1. **API:** The API is what users see when they first use
+   `napari`. APIs are difficult to change once released, so
+   should be simple, [functional](https://en.wikipedia.org/wiki/Functional_programming) (i.e. not
+   carry state), consistent with other parts of the library, and
+   should avoid modifying input variables.   
+
+2. **Documentation:** Any new feature should have a gallery
+   example, that not only illustrates but explains it.
+
+3. **The algorithm:** You should understand the code being modified or
+   added before approving it.  (See `Merge Only Changes You
+   Understand` below.) Implementations should do what they claim,
+   and be simple, readable, and efficient.
+
+4. **Tests:** All contributions  *must* be tested, and
+   each added line of code should be covered by at least one test. Good
+   tests not only execute the code, but explores corner cases.  It is tempting
+   not to review tests, but please do so.
+
+Other changes may be *nitpicky*: spelling mistakes, formatting,
+etc. Do not ask contributors to make these changes, and instead
+make the changes by [pushing to their branch](https://help.github.com/articles/committing-changes-to-a-pull-request-branch-created-from-a-fork/), or using GitHub’s [suggestion](https://help.github.com/articles/commenting-on-a-pull-request/) [feature]
+(https://help.github.com/articles/incorporating-feedback-in-your-pull-request/).
+(The latter is preferred because it gives the contributor a choice in
+whether to accept the changes.)
+
+Unless you know that a contributor is experienced with git, don’t
+ask for a rebase when merge conflicts arise. Instead, rebase the
+branch yourself, force-push to their branch, and advise the contributor to force-pull.  If the contributor is
+no longer active, you may take over their branch by submitting a new pull
+request and closing the original. In doing so, ensure you communicate
+that you are not throwing the contributor's work away!
+
+Please add a note to a pull request after you push new changes; GitHub
+does not send out notifications for these.
```

Comment:
I think the clarification here just needs to be "please add a note to a pull request when it is ready for another round of review". Having said this, GitHub now does do notifications for commits so that should also be edited, both here and in skimage's guide 😂

Reply:
I think we can drop the line, as it is targeted at the core dev, not the contributor, and agreed we get lots of notification these days!!